### PR TITLE
[docs] Model formats

### PR DESCRIPTION
**Pull request by stevhliu**
Refactors the `Model files and layouts` guide to from a more top-down approach, beginning with the formats (Diffusers/single-file) and then discussing the individual file types.

### CODE REVIEW

File: docs/source/en/using-diffusers/other-formats.md [hunk 1]
```diff
@@ -10,504 +10,294 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-## Files
-
-PyTorch model weights are typically saved with Python's [pickle](https://docs.python.org/3/library/pickle.html) utility as ckpt or bin files. However, pickle is not secure and pickled files may contain malicious code that can be executed. This vulnerability is a serious concern given the popularity of model sharing. To address this security issue, the [Safetensors](https://hf.co/docs/safetensors) library was developed as a secure alternative to pickle, which saves models as safetensors files.
-
-### safetensors
+Diffusion models are typically stored in the Diffusers format or single-file format. Model files can be stored in various file types such as safetensors, dduf, or ckpt.
 
 > [!TIP]
-> Learn more about the design decisions and why safetensor files are preferred for saving and loading model weights in the [Safetensors audited as really safe and becoming the default](https://blog.eleuther.ai/safetensors-security-audit/) blog post.
+> Format refers to the directory structure and file refers to the file type.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/other-formats.md]: Single file checkpoints might not have any directory structure, though.

File: docs/source/en/using-diffusers/other-formats.md [hunk 2]
```diff
@@ -10,504 +10,294 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+import torch
 from diffusers import DiffusionPipeline
 
-pipeline = DiffusionPipeline.from_pretrained(
-    "stable-diffusion-v1-5/stable-diffusion-v1-5",
-    use_safetensors=True
+pipeline = DiffusionPipeline.from_single_file(
+    "https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/sd_xl_base_1.0.safetensors",
+    torch_dtype=torch.float16,
+    device_map="cuda"
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/other-formats.md:49]: I don't think we support `device_map="cuda"` in `from_single_file` yet. Cc: @DN6 should we add support?

File: docs/source/en/using-diffusers/other-formats.md [hunk 3]
```diff
@@ -10,504 +10,294 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+from diffusers import DiffusionPipeline, DPMSolverMultistepScheduler
 
-pipeline = StableDiffusionPipeline.from_single_file(
-    "https://huggingface.co/WarriorMama777/OrangeMixs/blob/main/Models/AbyssOrangeMix/AbyssOrangeMix.safetensors"
+scheduler = DPMSolverMultistepScheduler()
+pipeline = DiffusionPipeline.from_single_file(
+    "https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/sd_xl_base_1.0.safetensors",
+    scheduler=scheduler,
+    torch_dtype=torch.float16,
+    device_map="cuda"
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/other-formats.md:66]: Same.

File: docs/source/en/using-diffusers/other-formats.md [hunk 4]
```diff
@@ -10,504 +10,294 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 ```
 
-</hfoption>
-</hfoptions>
-
-#### LoRAs
+### Configuration options
 
-[LoRAs](../tutorials/using_peft_for_inference) are lightweight checkpoints fine-tuned to generate images or video in a specific style. If you are using a checkpoint trained with a Diffusers training script, the LoRA configuration is automatically saved as metadata in a safetensors file. When the safetensors file is loaded, the metadata is parsed to correctly configure the LoRA and avoids missing or incorrect LoRA configurations.
+Models have a `config.json` file in their repositories with important attributes such as the number of layers and attention heads. The [`~loaders.FromSingleFileMixin.from_single_file`] method automatically determines the appropriate config to use from `config.json`.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/other-formats.md]: An important note here is that `config.json` is only present in Diffusers style checkpoints.

File: docs/source/en/using-diffusers/other-formats.md [hunk 5]
```diff
@@ -10,504 +10,294 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 
-The easiest way to inspect the metadata, if available, is by clicking on the Safetensors logo next to the weights.
-
-<div class="flex justify-center">
-    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/safetensors_lora.png"/>
-</div>
-
-For LoRAs that aren't trained with Diffusers, you can still save metadata with the `transformer_lora_adapter_metadata` and `text_encoder_lora_adapter_metadata` arguments in [`~loaders.FluxLoraLoaderMixin.save_lora_weights`] as long as it is a safetensors file.
+But if the models in a pipeline are different from the original implementation or if it doesn't have to necessary metadata to determine the correct config, then you need to use the `config` argument.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/other-formats.md]: This is one part. Another good point is that we try to automatically infer the appropriate config based on the checkpoint provided. While that works out in 99% of the cases (maybe 100%), if the users want to be explicit about this, they can always pass the config themselves.

File: docs/source/en/using-diffusers/other-formats.md [hunk 6]
```diff
@@ -10,504 +10,294 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+You could also load a config file not stored on the Hub by passing a local path or URL of the config file to the `original_config` argument.
 
 ```py
-from diffusers import StableDiffusionPipeline
+from diffusers import WanPipeline
 
-pipeline = StableDiffusionPipeline.from_single_file(
-    "https://huggingface.co/stable-diffusion-v1-5/stable-diffusion-v1-5/blob/main/v1-5-pruned.ckpt"
+ckpt_path = "https://huggingface.co/lightx2v/Qwen-Image-Lightning/blob/main/Qwen-Image-Lightning-8steps-V1.1-bf16.safetensors"
+original_config = "https://raw.githubusercontent.com/Wan-Video/Wan2.2/refs/heads/main/wan/configs/wan_ti2v_5B.py"
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/other-formats.md]: A Python file as a config? 💡

File: docs/source/en/using-diffusers/other-formats.md [hunk 7]
```diff
@@ -10,504 +10,294 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+    my_local_checkpoint_path, config=my_local_config_path, local_files_only=True
 )
-pipeline.save_pretrained()
 ```
 
-Lastly, there are also Spaces, such as [SD To Diffusers](https://hf.co/spaces/diffusers/sd-to-diffusers) and [SD-XL To Diffusers](https://hf.co/spaces/diffusers/sdxl-to-diffusers), that provide a more user-friendly interface for converting models to Diffusers-multifolder layout. This is the easiest and most convenient option for converting layouts, and it'll open a PR on your model repository with the converted files. However, this option is not as reliable as running a script, and the Space may fail for more complicated models.
+## File types
 
-## Single-file layout usage
+Models can be stored in several file types. Safetensors is the most common file type but you may encounter other file types on the Hub or diffusion community.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/other-formats.md:161]: Should `safetensors` be hyperlinked?

File: docs/source/en/using-diffusers/other-formats.md [hunk 8]
```diff
@@ -10,504 +10,294 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 
-Now that you're familiar with the differences between the Diffusers-multifolder and single-file layout, this section shows you how to load models and pipeline components, customize configuration options for loading, and load local files with the [`~loaders.FromSingleFileMixin.from_single_file`] method.
+### safetensors
 
-### Load a pipeline or model
+[Safetensors](https://hf.co/docs/safetensors) is a safe and fast file type for securely storing and loading tensors. It restricts the header size to limit certain types of attacks, supports lazy loading (useful for distributed setups), and generally loads faster.
 
-Pass the file path of the pipeline or model to the [`~loaders.FromSingleFileMixin.from_single_file`] method to load it.
+Diffusers loads safetensors file by default if they are available and the Safetensors library is installed.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/other-formats.md]: Safetenors is a required dependency in Diffusers, btw.

File: docs/source/en/using-diffusers/other-formats.md [hunk 9]
```diff
@@ -10,504 +10,294 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-from diffusers import StableDiffusionXLPipeline, DDIMScheduler
-
-ckpt_path = "https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/sd_xl_base_1.0_0.9vae.safetensors"
-scheduler = DDIMScheduler()
-pipeline = StableDiffusionXLPipeline.from_single_file(ckpt_path, scheduler=scheduler)
-```
+If you're using a checkpoint trained with a Diffusers training script, metadata such as the LoRA configuration, is automatically saved. When the file is loaded, the metadata is parsed to correctly configure the LoRA and avoid missing or incorrect LoRA configs. Inspect the metadata of a safetensors file by clicking on the <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/safetensors/logo.png" alt="safetensors logo" height="15em" style="vertical-align: middle;"> logo next to the file on the Hub.
 
-Or you could use a ControlNet model in the pipeline.
+Save the metadata for LoRAs that aren't trained with Diffusers with the `transformer_lora_adapter_metadata` and `text_encoder_lora_adapter_metadata` arguments in [`~loaders.FluxLoraLoaderMixin.save_lora_weights`]. This is only supported for safetensors files.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/other-formats.md]: For `unet`, it would be `unet_lora_adapter_metadata`.

For pipelines, incorporating multiple text encoders in LoRAs, we usually do:

```
text_encoder_lora_adapter_metadata
text_encoder_2_lora_adapter_metadata
```

**Review by sayakpaul — COMMENTED** [review]
Thank you!

File: docs/source/en/using-diffusers/other-formats.md (hunk 7 above)
Comment (stevhliu) [reply docs/source/en/using-diffusers/other-formats.md:161]: Hyperlinked in the Safetensors section below :)

File: docs/source/en/using-diffusers/other-formats.md (hunk 6 above)
Comment (stevhliu) [reply docs/source/en/using-diffusers/other-formats.md]: Oops, just realized Wan doesn't support `from_single_file`. Reverted to the original code example!

**Review by sayakpaul — APPROVED** [review]
@DN6 please give it a review as well.

File: docs/source/en/using-diffusers/other-formats.md (hunk 2 above)
Comment (DN6) [reply docs/source/en/using-diffusers/other-formats.md:49]: Supported now
https://github.com/huggingface/diffusers/blob/843355f89fd043e82b3344d9259e6faa640da6f9/src/diffusers/loaders/single_file_model.py#L423

Snippet looks good 👍🏽

File: docs/source/en/using-diffusers/other-formats.md [hunk 10]
```diff
@@ -10,504 +10,288 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-PyTorch's [torch.save](https://pytorch.org/docs/stable/generated/torch.save.html) function uses Python's [pickle](https://docs.python.org/3/library/pickle.html) utility to serialize and save models. These files are saved as a ckpt file and they contain the entire model's weights.
-
-Use the [`~loaders.FromSingleFileMixin.from_single_file`] method to directly load a ckpt file.
-
-```py
-from diffusers import StableDiffusionPipeline
-
-pipeline = StableDiffusionPipeline.from_single_file(
-    "https://huggingface.co/stable-diffusion-v1-5/stable-diffusion-v1-5/blob/main/v1-5-pruned.ckpt"
+pipeline = FluxPipeline.from_single_file(
```
Comment (DN6) [inline docs/source/en/using-diffusers/other-formats.md]: This won't work. Single file expects a URL  or path to a single safetensors file. Not a repo_id. Pipeline should use 

```suggestion
pipeline = FluxPipeline.from_pretrained(
```

File: docs/source/en/using-diffusers/other-formats.md [hunk 11]
```diff
@@ -10,504 +10,288 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-```
+A single-file format stores *all* the model (UNet, transformer, text encoder) weights in a single file. Benefits of single-file formats include the following.
 
-</hfoption>
-<hfoption id="single file">
+- Greater compatibility with [ComfyUI](https://github.com/comfyanonymous/ComfyUI) or [Automatic1111](https://github.com/AUTOMATIC1111/stable-diffusion-webui).
+- Easier to download and share a single file.
 
-Use the [`~loaders.FromSingleFileMixin.from_single_file`] method to load a model with all the weights stored in a single safetensors file.
+Use [`~loaders.FromSingleFileMixin.from_single_file`] to load a single file. Pass `"cuda"` to the `device_map` argument to pre-allocate GPU memory and reduce model loading time (refer to the [parallel loading](../using-diffusers/loading#parallel-loading) docs for more details).
```
Comment (DN6) [inline docs/source/en/using-diffusers/other-formats.md]: Single file does support pre-allocating GPU memory to reduce model loading time, but parallel loading only works for sharded checkpoints (one more spread over multiple files).

File: docs/source/en/using-diffusers/other-formats.md [hunk 12]
```diff
@@ -10,504 +10,288 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-
-1. Easy compatibility with diffusion interfaces such as [ComfyUI](https://github.com/comfyanonymous/ComfyUI) or [Automatic1111](https://github.com/AUTOMATIC1111/stable-diffusion-webui) which commonly use a single-file layout.
-2. Easier to manage (download and share) a single file.
-
-### DDUF
-
-> [!WARNING]
-> DDUF is an experimental file format and APIs related to it can change in the future.
+ckpt_path = "https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/sd_xl_base_1.0_0.9vae.safetensors"
+original_config = "https://raw.githubusercontent.com/Stability-AI/generative-models/main/configs/inference/sd_xl_base.yaml"
```
Comment (DN6) [inline docs/source/en/using-diffusers/other-formats.md]: Original config is only meant to support SD/SDXL models config provided as yaml files. It will be deprecated with Diffusers 1.0.0

IMO better not to highlight or include a note saying this will be deprecated.

File: docs/source/en/using-diffusers/other-formats.md [hunk 13]
```diff
@@ -10,504 +10,288 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+    my_local_checkpoint_path, config=my_local_config_path, local_files_only=True
+)
+```
 
-Pass the file path of the pipeline or model to the [`~loaders.FromSingleFileMixin.from_single_file`] method to load it.
+### Symlink
 
-<hfoptions id="pipeline-model">
-<hfoption id="pipeline">
+If you're working with a file system that does not support symlinking, download the checkpoint file to a local directory first. Disable symlinking with `local_dir_use_symlink=False` in [`~huggingface_hub.snapshot_download`] and [`~huggingface_hub.hf_hub_download`].
```
Comment (DN6) [inline docs/source/en/using-diffusers/other-formats.md]: I think the hub team removed this requirement. If you specify local_dir, then it automatically doesn't use symlinks.

File: docs/source/en/using-diffusers/other-formats.md [hunk 14]
```diff
@@ -10,504 +10,288 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-
-Customize components in the pipeline by passing them directly to the [`~loaders.FromSingleFileMixin.from_single_file`] method. For example, you can use a different scheduler in a pipeline.
-
-```py
-from diffusers import StableDiffusionXLPipeline, DDIMScheduler
+my_local_checkpoint_path = hf_hub_download(
+    repo_id="segmind/SSD-1B",
+    filename="SSD-1B.safetensors"
+    local_dir="my_local_checkpoints",
+    local_dir_use_symlinks=False
```
Comment (DN6) [inline docs/source/en/using-diffusers/other-formats.md]: Can remove `local_dir_use_symlinks=False`

File: docs/source/en/using-diffusers/other-formats.md [hunk 15]
```diff
@@ -10,504 +10,288 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+)
+print("My local checkpoint: ", my_local_checkpoint_path)
 
-ckpt_path = "https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/sd_xl_base_1.0_0.9vae.safetensors"
-scheduler = DDIMScheduler()
-pipeline = StableDiffusionXLPipeline.from_single_file(ckpt_path, scheduler=scheduler)
+my_local_config_path = snapshot_download(
+    repo_id="segmind/SSD-1B",
+    allow_patterns=["*.json", "**/*.json", "*.txt", "**/*.txt"]
+    local_dir_use_symlinks=False,
```
Comment (DN6) [inline docs/source/en/using-diffusers/other-formats.md]: Can remove `local_dir_use_symlinks=False`

File: docs/source/en/using-diffusers/other-formats.md (hunk 11 above)
Comment (stevhliu) [reply docs/source/en/using-diffusers/other-formats.md]: I thought with https://github.com/huggingface/diffusers/pull/12305, it is supported for `from_single_file` now?

Comment (DN6) [reply docs/source/en/using-diffusers/other-formats.md]: Parallel loading only works for multiple files (sharded checkpoints) since you load them simultaneously. Since single file is just one file, you won't see any advantage.

Comment (stevhliu) [reply docs/source/en/using-diffusers/other-formats.md]: Ah, I see! Ok, removed mention of pre-allocating :)
